rocksdb_admin/admin_handler.cpp

```diff
@@ -1040,25 +1047,43 @@ void AdminHandler::async_tm_startMessageIngestion(
       kafka_consumer_pool,
       -1); // kafka_init_blocking_consumer_timeout_ms ;
 
-  {
-    std::lock_guard<std::mutex> lock(kafka_watcher_lock_);
-    kafka_watcher_map_[db_name] = kafka_watcher;
-  }
+  auto kafka_spec = kafka_spec_manager_.addSpec(db_name, kafka_watcher,
+      meta.last_kafka_msg_timestamp_ms);
+
+  kafka_ts_updater_.addFunction(
+    [db_name, kafka_spec, this] {
```

Comment:
Did we consider put this logic into the lambda that passed to StartWith() below? that will make the code simpler.
The downside is that timestamp persistence can only be triggered when a new message is received. That should be fine?

Reply:
Yes, I did consider it. I preferred this since we don't have to do the time management. We would still need a global timestamp per db, or read the meta_db for every message consumed. Since this is more of a background task, and it's not super critical to run at the exact interval, thought it would be simpler to do all that in a separate thread.

Reply:
We can capture an int or a timestamp in the lambda, and for each message received, we bump the int (if we want to persist the timestamp once every N message). Or we update the captured timestamp whenever we persist a timestamp to db (if we want to persist the timestamp once every X mins). Will it work?

This will make the code much simpler. we don't need KafkaSpecs related classes any more.

Reply:
Sounds good, thanks for the suggestion. I'll close this PR and create a new one.